### DISCUSSION TITLE
Not authorized to program in library when debugging

### DISCUSSION THREAD

**Discussion by BastianZwetsch** [codefori/vscode-ibmi#1201] [General]
I have been trying to use the new IBM i Debug plugin for Code for IBM i for a while now and i cant seem to get it to work. 
Everytime I try to debug a program i get a popup-window which says "You are not authorized to program [Programname] in library [libraryname]" with the options to open "launch.json", which I dont have,  or "Cancel", which just closes the editor. Here is an example:
![image](https://user-images.githubusercontent.com/106727583/228785609-5519c05b-8f4d-49cf-8125-78e29ae2497d.png)
I am relativly new to the IBM i so i was out of ideas pretty fast.
The debugger whe have been using on the IBM i is pretty bad so it would help immensly if we get this to work.

Thanks in advance!

**Comment by sebCIL** [comment]
Hello,

I have the same problem with programs whose sources are not in the same library.
